### PR TITLE
Add more WasmEngine tests

### PR DESCRIPTION
**Pull request by axic**

### CODE REVIEW

File: test/unittests/wasm_engine_test.cpp [hunk 1]
```diff
@@ -5,6 +5,63 @@
...
+    const auto wasm = "0102"_bytes;
+
+    for (auto engine_create_fn : {create_fizzy_engine, create_wabt_engine, create_wasm3_engine})
+    {
+        auto engine = engine_create_fn();
+        ASSERT_FALSE(engine->parse(wasm));
+    }
+}
+
+TEST(wasm_engine, DISABLED_instantiate_error)
```
Comment (chfast) [inline test/unittests/wasm_engine_test.cpp]: Why is this disabled?

Comment (axic) [reply test/unittests/wasm_engine_test.cpp]: Need to fix #138 to make this pass on wasm3. And wabt had some other issue I can't remember.

Comment (axic) [reply test/unittests/wasm_engine_test.cpp]: I can enable this test for fizzy only, if that helps.

Comment (chfast) [reply test/unittests/wasm_engine_test.cpp]: I'm not sure we can have generically parse and instantiate separated. You can leave it for fizzy only.

**Pull request merged**